mdbtools compile failure on RHEL 7.9 (with patch to fix)

This is a patch to mdbtools for Red Hat Enterprise Linux Server release 7.9 (Maipo).
The fix is to add headers and a define to mdbsql.c .

Warnings are being treated as errors, and it throws this error message in the make step on my RHEL 7.
 
```
make[3]: Entering directory `/hfs/d1/roq300/tools/mdbtools-0.9.3/src/sql'
  CC       mdbsql.lo
mdbsql.c: In function 'mdb_sql_error':
mdbsql.c:59:2: error: implicit declaration of function 'vsnprintf' [-Werror=implicit-function-declaration]
  vsnprintf(sql->error_msg, sizeof(sql->error_msg), fmt, ap);
  ^
mdbsql.c: In function 'mdb_sql_strptime':
mdbsql.c:377:3: error: implicit declaration of function 'snprintf' [-Werror=implicit-function-declaration]
   snprintf(pszDate, 16, "%lf", date);
   ^
mdbsql.c:377:3: error: incompatible implicit declaration of built-in function 'snprintf' [-Werror]
mdbsql.c: In function 'mdb_sql_add_sarg':
mdbsql.c:453:3: error: incompatible implicit declaration of built-in function 'snprintf' [-Werror]
   snprintf(node->value.s, sizeof(node->value.s), "%.*s", (int)strlen(constant) - 2, &constant[1]);
   ^
mdbsql.c: In function 'mdb_sql_describe_table':
mdbsql.c:686:3: error: incompatible implicit declaration of built-in function 'snprintf' [-Werror]
   snprintf(tmpstr, sizeof(tmpstr), "%s", mdb_get_colbacktype_string(col));
   ^
mdbsql.c: In function 'mdb_sql_select':
mdbsql.c:786:3: error: incompatible implicit declaration of built-in function 'snprintf' [-Werror]
   snprintf(tmpstr, sizeof(tmpstr), "%d", table->num_rows);
   ^
cc1: all warnings being treated as errors
```
 
 
Here is the patch, first cd to src/sql, I use an environment variable for where I am doing the build:
```cd $MDBTOOLS_VER/src/sql```
 
This is an old-school patch file
 
```
cat > patch_xopen <<EOF
*** mdbsql.c    Sun Apr 25 08:11:19 2021
--- ../sql_patch_xopen/mdbsql.c Sat May  8 11:39:00 2021
***************
*** 16,21 ****
--- 16,25 ----
   * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
   */
 
+ #include <stdio.h>
+ #define __USE_XOPEN
+ #include <time.h>
+
  #include <stdarg.h>
  #ifndef _XOPEN_SOURCE
  #define _XOPEN_SOURCE
EOF
 
# Apply the patch
patch mdbsql.c patch_xopen
 
# Get rid of the patch file, it is no longer needed
rm patch_xopen
```

--- a/src/sql/mdbsql.c
+++ b/src/sql/mdbsql.c
@@ -18,7 +18,7 @@
 
 #include <stdarg.h>
 #ifndef _XOPEN_SOURCE
-#define _XOPEN_SOURCE
+#define _XOPEN_SOURCE 600
 #endif
 #include "mdbsql.h"
 

